File: src/FreeSqlBuilderUI/Client/FreeSqlBuilderClient/ClientApp/src/app/project/modals/generatormodeconfig.ts

export enum DataType {
    MySql = 0,
    SqlServer = 1,
    PostgreSQL = 2,
    Oracle = 3,
    Sqlite = 4,
    OdbcOracle = 5,
    OdbcSqlServer = 6,
    OdbcMySql = 7,
    OdbcPostgreSQL = 8,
    Odbc = 9,
    OdbcDameng = 10,
    MsAccess = 11
}
export class DataSource {
    constructor(public id = 0, public name = ``, public dbType = DataType.SqlServer, public connectionString = ``
    ) {

    }
}

export enum PickType {
    Pick, Ignore
}
export class GeneratorModeConfig {
    constructor(projectid: number) {
        this.projectId = projectid;
        this.generatorMode = 1;
        this.id = 0;
        this.entityAssemblyName = '';
        this.entityBaseName = '';
        this.ignoreTables = '';
        this.includeTables = '';
        this.pickType = PickType.Ignore;
    }
    id: number;
    projectId: number;
    generatorMode: GeneratorMode = 1;
    dataSource: DataSource = new DataSource();
    entityAssemblyName: string;
    dataSourceId: number;
    includeTables: string;
    ignoreTables: string;
    entityBaseName: string;
    pickType: PickType = PickType.Ignore;
}
export class TableInfo {
    isServiceTable: boolean;
    nameSpace: string;
    importUsings: string[];
    primaryTypeName: string;
    comment: string;
    name: string;
    dbTableName: string;
    columnInfos: ColumnInfo[] = new Array<ColumnInfo>();
    navigateInfos: NavigateColumnInfo[] = new Array<NavigateColumnInfo>();
}

export class ColumnInfo {
    columnName: string;
    comment: string;
    dbColumnName: string;
    dbType: string;
    csType: string;
}

// tslint:disable-next-line: no-use-before-declare
export class NavigateColumnInfo extends ColumnInfo {
    dbColumnName: string;
    navigateTableInfo: TableInfo;
    navigateCategory: NavigateCategory;
}


export enum NavigateCategory {
    OneToOne,
    OneToMany,
    None
}

export enum GeneratorMode {
    DbFirst,
    CodeFirst
}

File: src/FreeSqlBuilder/FreeSqlBuilder.Core/GeneratorModeConfig.cs
using System.Collections.Generic;
using System.Linq;
using FreeSql.DataAnnotations;

namespace FreeSqlBuilder.Core
{
    public class GeneratorModeConfig : IKey<long>
    {
        [Column(IsPrimary = true, IsIdentity = true)]
        public long Id { get; set; }
        /// <summary>
        /// 项目
        /// </summary>
        public Project Project { get; set; }
        /// <summary>
        /// 项目id
        /// </summary>
        [Column(IsNullable = false)]
        public long ProjectId { get; set; }
        /// <summary>
        /// 实体
        /// </summary>
        public string EntityAssemblyName { get; set; }
        /// <summary>
        /// 生成器模式 CodeFirst根据实体代码来生成 DbFirst根据数据库生成代码
        /// </summary>
        [Column(IsNullable = false)]
        public GeneratorMode GeneratorMode { get; set; } = GeneratorMode.CodeFirst;
        /// <summary>
        /// 基类
        /// </summary>
        public string EntityBaseName { get; set; }
        /// <summary>
        /// 数据库信息
        /// </summary>
        public DataSource DataSource { get; set; }
        /// <summary>
        /// 数据源id
        /// </summary>
        public long DataSourceId { get; set; }
        /// <summary>
        /// 选中模式
        /// </summary>
        public PickType PickType { get; set; }

        /// <summary>
        /// 只生成某些表
        /// </summary>
        public string IncludeTables { get; set; }
        /// <summary>
        /// 忽略某些表
        /// </summary>
        public string IgnoreTables { get; set; }
        /// <summary>
        /// 只生成某些表
        /// </summary>
        [Column(IsIgnore = true)]
        public List<string> IncludeTable => IncludeTables?.Split(",").ToList();
        /// <summary>
        /// 忽略某些表
        /// </summary>
        [Column(IsIgnore = true)]
        public List<string> IgnoreTable => IgnoreTables?.Split(",").ToList();

    }

    public enum PickType
    {
        Pick, Ignore
    }
}
